[verbs] compiler warning

Got this compiler warning from the verbs provider this morning:

```
In file included from prov/verbs/src/verbs_cq.c:35:0:
prov/verbs/src/verbs_cq.c: In function ‘fi_ibv_cq_readerr’:
./include/fi_mem.h:198:2: warning: ‘slist_entry’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  slist_insert_head(&util_buf->entry, &pool->buf_list);
  ^
prov/verbs/src/verbs_cq.c:46:22: note: ‘slist_entry’ was declared here
  struct slist_entry *slist_entry;
                      ^
```

--- a/man/fi_cq.3.md
+++ b/man/fi_cq.3.md
@@ -22,6 +22,9 @@ fi_cq_sread / fi_cq_sreadfrom
 : A synchronous (blocking) read that waits until a specified condition
   has been met before reading a completion from a completion queue.
 
+fi_cq_signal
+: Unblock any thread waiting in fi_cq_sread or fi_cq_sreadfrom.
+
 fi_cq_strerror
 : Converts provider specific error information into a printable string
 

--- a/prov/verbs/src/verbs_cq.c
+++ b/prov/verbs/src/verbs_cq.c
@@ -48,15 +48,14 @@ fi_ibv_cq_readerr(struct fid_cq *cq_fid, struct fi_cq_err_entry *entry,
 	cq = container_of(cq_fid, struct fi_ibv_cq, cq_fid);
 
 	fastlock_acquire(&cq->lock);
-	if (!slist_empty(&cq->wcq)) {
-		wce = container_of(cq->wcq.head, struct fi_ibv_wce, entry);
-		if (!wce->wc.status) {
-			fastlock_release(&cq->lock);
-			return -FI_EAGAIN;
-		}
+	if (slist_empty(&cq->wcq))
+		goto err;
 
-		slist_entry = slist_remove_head(&cq->wcq);
-	}
+	wce = container_of(cq->wcq.head, struct fi_ibv_wce, entry);
+	if (!wce->wc.status)
+		goto err;
+
+	slist_entry = slist_remove_head(&cq->wcq);
 	fastlock_release(&cq->lock);
 
 	wce = container_of(slist_entry, struct fi_ibv_wce, entry);
@@ -70,6 +69,9 @@ fi_ibv_cq_readerr(struct fid_cq *cq_fid, struct fi_cq_err_entry *entry,
 
 	util_buf_release(cq->domain->fab->wce_pool, wce);
 	return sizeof(*entry);
+err:
+	fastlock_release(&cq->lock);
+	return -FI_EAGAIN;
 }
 
 static inline int

